tests/test_downloader.py

```
@@ -3731,6 +3843,217 @@ def test_error_handling_comprehensive_ui_paths(tmp_path, caplog):
+        ]
+
+        # Mock successful download
+        def mock_download_func(url, dest):
+            Path(dest).parent.mkdir(parents=True, exist_ok=True)
+            Path(dest).write_bytes(b"fake_firmware_data")
+            return True
+
+        mock_download.side_effect = mock_download_func
+
+        # Call check_for_prereleases
+        result = check_for_prereleases(
```

Ruff (0.13.1)

Local variable `result` is assigned to but never used

Remove assignment to unused variable `result`

(F841)